Add jobs for jdk8 and jdk11

schannamallu opened the pull request:
Signed-off-by: Suresh Channamallu <schannamallu@linuxfoundation.org>

Add jobs for jdk8 and jdk11 for project odpi-egeria, egeria-connector-ibm-igc
Modify jenkins-config/clouds/openstack/Primary/cloud.cfg with latest image.

#62

schannamallu committed:
Add jobs for jdk8 and jdk11

Signed-off-by: Suresh Channamallu <schannamallu@linuxfoundation.org>

jwagantall commented on jjb/egeria/egeria.yaml:
```diff
@@ -10,6 +10,9 @@
     stream:
       - 'master':
           branch: master
+    java-version:
```
Is this supported? 2 JDKs?

schannamallu replied:
> I am not sure the commit text matches the file changes since we are not creating new jobs.
> 
> Can we please fix that?

By this change there will be a new jobs generated - odpi-egeria-maven-verify-master-mvn35-openjdk11, egeria-connector-ibm-igc-maven-verify-master-mvn35-openjdk11. 
Do I need to change the commit message to generate.. ??

schannamallu replied:
> Is this supported? 2 JDKs?

Yes, With the new Image it's supported. Tested them in sandbox.

jwagantall reviewed (changes requested):
I am not sure the commit text matches the file changes since we are not creating new jobs. 

Can we please fix that?

tykeal reviewed (dismissed):
The change to the cloud.cfg is not related to the change in jobs. At least not directly. That should be a completely separate change

schannamallu commented:
> The change to the cloud.cfg is not related to the change in jobs. At least not directly. That should be a completely separate change

Actually, With out the cloud.cfg change (Selecting new image) the jdk11 jobs wont work (Older/Current image dont have jdk11 installed).
For fixing the issue #62 I felt this both changes are relevant.

tykeal commented:
> > The change to the cloud.cfg is not related to the change in jobs. At least not directly. That should be a completely separate change
> 
> Actually, With out the cloud.cfg change (Selecting new image) the jdk11 jobs wont work (Older/Current image dont have jdk11 installed).
> For fixing the issue #62 I felt this both changes are relevant.

What I'm trying to get at is that you're modifying the base config. Yes that needs to be updated, but you also _must_ update the config associated with a given minion definition or your definitions aren't going to be running on the image you want. The config job detects differences between the base config and the minion definition, and if they are defined differently, then the defined minion will get a custom image applied, if they are defined the same, the minion will get an 'inherit from base' defined.

schannamallu commented:
> > > The change to the cloud.cfg is not related to the change in jobs. At least not directly. That should be a completely separate change
> > 
> > 
> > Actually, With out the cloud.cfg change (Selecting new image) the jdk11 jobs wont work (Older/Current image dont have jdk11 installed).
> > For fixing the issue #62 I felt this both changes are relevant.
> 
> What I'm trying to get at is that you're modifying the base config. Yes that needs to be updated, but you also _must_ update the config associated with a given minion definition or your definitions aren't going to be running on the image you want. The config job detects differences between the base config and the minion definition, and if they are defined differently, then the defined minion will get a custom image applied, if they are defined the same, the minion will get an 'inherit from base' defined.

Updated the existing commit and added new PR

jwagantall reviewed (commented):
Nevermind.. was looking at an old patchset..
